caldav/server.go

```diff
@@ -42,6 +40,11 @@ type Backend interface {
 	webdav.UserPrincipalBackend
 }
 
+type MultiCalendarBackend interface {
```

Comment:
I'm not familiar enough with `go` to know if this is a reasonable way to implement this change. This means we're 100% backwards compatible, and it's up for the user of the library to conform to `MulitCalendarBackend` to get that support, I think the biggest drawback to this is that discoverability suffers, and it adds some more possible paths for the code to take here. Maybe it would be OK to break the API instead, adopting the changes should be very straightforward (just return a `[]*Calendar` instead of `*Calendar`) etc.